### ISSUE TITLE
Control network configurations for Docker Desktop Business

### ISSUE THREAD

**Issue by voliveirajr** [docker/roadmap#284]
**Tell us about your request**
Ability to restrict what networks Docker Client and associated containers can connect to i.e. constrain the client to it's own network. Specifically prevent outbound access to internet.

**Which service(s) is this request for?**
Docker Desktop Business

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
A corporate client wants to have control what network users can use limiting it to a pre-defined set

**Are you currently working around the issue?**
Was considered restrict it by AuthZ plugin however it's a bit more complex when dealing with dockerfile and docker-compose.
No workarounds at the moment.

**Additional context**
N/A

**Comment by stephanierifai** [comment]
Hi @voliveirajr I am curious, when you say a pre-defined set, are you imagining something in the Docker Dashboard local network settings that has a drop down with multiple options that are set by an admin? Or something else?
<img width="1116" alt="Screen Shot 2021-12-07 at 2 48 05 PM" src="https://user-images.githubusercontent.com/85884197/145096227-b724576d-5067-4b06-88c6-f9ae98ded755.png">

**Comment by voliveirajr** [comment]
@stephanierifai yes, I was thinking in let the administrator create the networks and let the user choose from a list or enforce the use of a predefined one. The most important here is to prevent the user to make use of the defaults bridge and host ones and create customised with 'docker network create'.
The administrator must have the option to:

- create new default network
- customised what networks the user can add to the containers
- Prevent the user to create/change networks

**Comment by stephanierifai** [comment]
Thank you @voliveirajr !

**Comment by christophermclellan** [comment]
Hey @voliveirajr, we're shipping Admin Controls for Docker Business customers in v4.13 of Desktop with the ability to pre-configure the Network setting that Steph sent the screenshot of above. The JSON config for this field will look like this:

```
   "vpnkitCIDR": {
      "locked": true,
      "value":"192.168.65.0/24"
    }
```

Is there anything you would change here ?
